### PR TITLE
create_test_village avoids constraint errors

### PR DESCRIPTION
**Pull request by weilu**
Also check first if the village with the given code already exists

@delcroip kindly review please.

### CODE REVIEW

File: location/test_helpers.py [hunk 1]
```diff
@@ -45,14 +51,39 @@ def create_test_location(loc_type, valid=True, custom_props=None):
...
     
-def create_test_village(custom_props=None):
-    if custom_props is None:
-        custom_props = {}
-    custom_props["name"]=custom_props['name'] if "name" in custom_props else "Test Village"
-    test_region = create_test_location('R',custom_props={"name": "Region" + custom_props["name"]})
-    test_district = create_test_location('D', custom_props={"parent": test_region,"name":"District " + custom_props["name"]})
-    test_ward = create_test_location('W', custom_props={"parent": test_district,"name":"Ward" + custom_props["name"]})
-    custom_props["parent"]=test_ward
+def create_test_village(custom_props={}):
```
Comment (delcroip) [inline location/test_helpers.py]: You cannot do that, in python default parameters are initialize once, for mutable it means that they might keep a value between call, yes it is weird but I saw it happening myself this week

**Review by delcroip — CHANGES_REQUESTED** [review]

File: location/test_helpers.py (hunk 1 above)
Comment (weilu) [reply location/test_helpers.py]: You are right. Good catch. Fixed.

**Comment by weilu** [comment]
@delcroip can you kindly review again?

**Review by delcroip — APPROVED** [review]

**Comment by weilu** [comment]
@delcroip can we merge this? I don't have permission.

**Pull request merged**